pkg/k8s/wrapper.go

```diff
@@ -220,3 +224,59 @@ func (k *k8sWrapper) ListEvents(ops []client.ListOption) (*eventsv1.EventList, e
 	}
 	return events, nil
 }
+
+func (k *k8sWrapper) GetCNINode(cniNodeName, cniNodeNamespace string) (*rcv1alpha1.CNINode, error) {
+	cninode := &rcv1alpha1.CNINode{}
+	if err := k.cacheClient.Get(k.context, types.NamespacedName{
+		Name:      cniNodeName,
+		Namespace: cniNodeNamespace,
+	}, cninode); err != nil {
+		return cninode, err
+	}
+	return cninode, nil
+}
+
+func (k *k8sWrapper) CreateCNINode(node *v1.Node) error {
+	err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			return true
+		},
+		func() error {
+			cniNode := &rcv1alpha1.CNINode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      node.Name,
+					Namespace: config.KubeDefaultNamespace,
+					// use the node as owner reference to let k8s clean up the CRD when the node is deleted
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: node.APIVersion,
+							Kind:       node.Kind,
+							Name:       node.Name,
+							UID:        node.UID,
+						},
+					},
+				},
+			}
+
+			// TODO: need think more if we should retry on "already exists" error.
+			return client.IgnoreAlreadyExists(k.cacheClient.Create(k.context, cniNode))
+		})
+
+	// do not return error if the error is "already existing"
+	// add a wrapper here to prevent we use retry on the error above
+	return client.IgnoreAlreadyExists(err)
+}
+
+func (k *k8sWrapper) DeleteCNINode(cniNodeName, cniNodeNamespace string) error {
+	err := retry.OnError(retry.DefaultBackoff, func(err error) bool { return true }, func() error {
```

Comment:
Same question for retrying here.

Reply:
This func has been removed since we won't want to add redundant garbage collector.